Sources/SwiftKuery/Connection.swift

```diff
@@ -124,59 +197,108 @@ public protocol Connection {
     func rollback(to savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))
 
+    /// Rollback the current transaction to the specified savepoint.
+    ///
+    /// - Parameter to savepoint: The name of the savepoint to rollback to.
+    /// - Returns: A QueryResult representing the result of the operation
+    func rollbackSync(to savepoint: String) -> QueryResult
+
     /// Release a savepoint.
     ///
     /// - Parameter savepoint: The name of the savepoint to release.
     /// - Parameter onCompletion: The function to be called when the execution of release savepoint command has completed.
-    func release(savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))    
+    func release(savepoint: String, onCompletion: @escaping ((QueryResult) -> ()))
+
+    /// Release a savepoint.
+    ///
+    /// - Parameter savepoint: The name of the savepoint to release.
+    /// - Returns: A QueryResult representing the result of the operation
+    func releaseSync(savepoint: String) -> QueryResult
 }
 
 public extension Connection {
+
     func execute(query: Query, parameters: [String:Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        do {
-            let databaseQuery = try query.build(queryBuilder: queryBuilder)
-            let (convertedQuery, namedToNumbered, count) = Utils.convertNamedParametersToNumbered(query: databaseQuery, queryBuilder: queryBuilder)
-            var numberedParameters: [Any?] = Array(repeating: nil, count: count)
-            for (parameterName, parameterValue) in parameters {
-                if let numbers = namedToNumbered[parameterName] {
-                    for number in numbers {
-                        numberedParameters[number - 1] = parameterValue
+        DispatchQueue.global().async {
+            do {
+                let databaseQuery = try query.build(queryBuilder: self.queryBuilder)
+                let (convertedQuery, namedToNumbered, count) = Utils.convertNamedParametersToNumbered(query: databaseQuery, queryBuilder: self.queryBuilder)
+                var numberedParameters: [Any?] = Array(repeating: nil, count: count)
+                for (parameterName, parameterValue) in parameters {
+                    if let numbers = namedToNumbered[parameterName] {
+                        for number in numbers {
+                            numberedParameters[number - 1] = parameterValue
+                        }
+                    }
+                    else {
+                        onCompletion(.error(QueryError.syntaxError("Failed to map parameters.")))
+                        return
                     }
                 }
-                else {
-                    onCompletion(.error(QueryError.syntaxError("Failed to map parameters.")))
-                }
+                self.execute(convertedQuery, parameters: numberedParameters, onCompletion: onCompletion)
```

Comment:
Check that this offloads again and if so update to only offload completion handlers calls.

Reply:
It does offload and have made changes accordingly.